public/app/plugins/datasource/loki/querybuilder/parsing.ts

```diff
@@ -526,9 +526,12 @@ function handleVectorAggregation(expr: string, node: SyntaxNode, context: Contex
   const params = [];
 
   const numberNode = node.getChild(NumberLezer);
+  const errorNode = node.getChild(ErrorId)?.getChild(Identifier);
 
   if (numberNode) {
     params.push(Number(getString(expr, numberNode)));
+  } else if (errorNode) {
+    params.push(getString(expr, errorNode));
   }
 
```

Comment:
I am a bit confused by this. Do you mind explaining this change @matyax?

Reply:
Sure thing. Maybe it's worth to add a comment explaining why. On a normal query, we read the contents of the Number node, if exists.

<img width="1009" alt="number" src="https://github.com/user-attachments/assets/897e562c-f00c-4c21-90aa-0ab8a193f7df" />

When the query has variables, they get parsed as errors, and in this case what we need is an Identifier node inside an error node.

<img width="716" alt="error" src="https://github.com/user-attachments/assets/58250373-73cb-4199-b55a-0c250e95497d" />

Reply:
Ah, I was under the impression `interpolateVariablesInQueries` would always create "valid" queries. Makes sense with that explanation.

Maybe worth taking a look at `interpolateVariablesInQueries` at one point too see if that can be improved.

Reply:
Yeah, it does. This change is only for the visual query builder, that uses ´replaceVariables´, creating queries like the one in the screenshots.

https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/loki/querybuilder/parsingUtils.ts#L38